Breadcrumbs.py

```diff
@@ -32,80 +38,169 @@ def get_row_indentation(points, view, tab_size, limit=1e20):
 def is_white_row(view, points):
   return all(view.substr(pt).isspace() for pt in reversed(points))
 
-def get_breadcrumb(view, points, limit):
+def get_breadcrumb(view, points, regex, separator):
   for pt in points:
     ch = view.substr(pt)
     if not ch.isspace():
-      return view.substr(sublime.Region(pt, min(view.line(pt).b, pt + limit))).strip()
+      linestring = view.substr(sublime.Region(pt, min(view.line(pt).b, pt + 500))).strip()
+      match = re.search(re.compile(regex), linestring)
+      if match:
+        return(separator + match.group('name'))
   return ''
```

Comment:
Simplest fix for now was to do 

```
if not this_breadcrumb == '':
        breadcrumbs.append(this_breadcrumb)
```

Your suggestion is more elegant, so perhaps I'll revisit this later.